sql/core/src/test/scala/org/apache/spark/sql/execution/BenchmarkWholeStageCodegen.scala

```diff
@@ -153,16 +153,36 @@ class BenchmarkWholeStageCodegen extends SparkFunSuite {
   ignore("aggregate with keys") {
     val N = 20 << 20
 
-    runBenchmark("Aggregate w keys", N) {
-      sqlContext.range(N).selectExpr("(id & 65535) as k").groupBy("k").sum().collect()
+    val benchmark = new Benchmark("Aggregate w keys", N)
```

Comment:
Extended `testFallbackStartsAt` to now support 2 offsets (for aggregate hash map and bytes to bytes map) and modified `AggregationQuerySuite` such that all existing tests test for all new codepaths.